components/grade-table.js

```diff
@@ -19,4 +23,11 @@ class GradeTable {
     }
     return totalGrade/grades.length;
   }
+  destroy() {
+    var tbodyElement = document.querySelector("tbody");
+    var currentRow = document.getElementsByClassName("dataRow");
+    for (var j = tbodyElement.childNodes.length - 1; j >= 0; j--) {
+        currentRow[j].remove();
+    }
```

Comment:
- You don't need to query the `"dataRow"` class elements. Just call `remove` on the `childNodes` directly:
    ```js
    tbodyElement.childNodes[j].remove();
    ```
- The `length` of `childNodes` gets shorter by one every time you `remove` a child node, which means you'll skip nodes if you iterate over it forward.